### PR TITLE
chore(deps): bump `@metamask/{swaps, gas-fee}-controller`

### PR DESCRIPTION
**Pull request by mikesposito**
## **Description**

 
This PR bumps `@metamask/swaps-controller` to `^11.0.0` and `@metamask/gas-fee-controller` to `^21.0.0` (to match swaps-controller's peer dependency requirements). See package changelogs for details on changes:
- [swaps-controller](https://github.com/MetaMask/swaps-controller/blob/main/CHANGELOG.md)
- [gas-fee-controller](https://github.com/MetaMask/core/blob/main/packages/gas-fee-controller/CHANGELOG.md)

## **Related issues**

Related to: https://github.com/MetaMask/metamask-mobile/issues/11910

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -189,7 +189,7 @@
     "@metamask/snaps-utils": "^8.1.1",
     "@metamask/stake-sdk": "^0.2.13",
     "@metamask/swappable-obj-proxy": "^2.1.0",
-    "@metamask/swaps-controller": "^10.0.0",
+    "@metamask/swaps-controller": "^11.0.0",
```
Comment (tommasini) [inline package.json:192]: In the changelog says it have the gas fee controller v21 as peer dependency, isn't that a blocker for this update?

Comment (mcmire) [reply package.json:192]: It looks like `swaps-controller` v10 has no peer dependency on `gas-fee-controller` — that was added in v11. So yes it does seem like we ought to bump `gas-fee-controller` first if we can.

Comment (mikesposito) [reply package.json:193]: Done in [3bd401f](https://github.com/MetaMask/metamask-mobile/pull/12368/commits/3bd401f088d0ed7211a6167a7ea3b691e88a0de8)!

**Review by Gudahtt — APPROVED** [review]
LGTM! Looks like no patches are affected, and this removes some peer dependency warnings. None of the breaking changes in these ranges appear to require changes either.

**Review by sethkfman — APPROVED** [review]
LGTM. Glad to see these type-expect-error ignore disappear

**Pull request merged**